Add file generator

jorinvo opened the pull request:
Takes a filepath and reads random lines. See #34.

jorinvo commented on pkg/fakedata/generator.go:
```diff
@@ -177,6 +178,27 @@ var enum = func(column Column) string {
 
+var fileCache map[string][]string
+
+var file = func(column Column) string {
```
@lucapette is there a reason you put the functions in `var`s instead of the usual `func file() ...`?

lucapette replied:
It's funny. I think it's literally "historical reasons" even though the project is so young. I started it off an existing codebase and evolved the design in a direction I liked more. So long story short, there's no good reason right now. I'll put in a small gardening/good first patch issue for it.

I think it's good we keep it consistent right now though so I appreciate you followed existing style.

lucapette replied:
#38

lucapette commented:
oh nice. I actually got on the laptop because I wanted to start working on this feature. It's so cool to see we're already off a nice start!

A couple of notes:

- I think the feature deserves integration tests. There's more than one thing that can go wrong (the file doesn't exist for example) and the use case is "a bit special"
- It's a pretty cool feature so it deserves some "special place" in the README

We could focus on the first part only for now as documentation is tricker than it looks like and I'd like a couple of days to think the README table of contents (we're about to add a lot of cool features!)

jorinvo commented:
@lucapette you're right. I forgot about those parts. But I won't continue tonight.
I can look into it tomorrow or you can continue, if you like.

lucapette commented on pkg/fakedata/generator.go:
```diff
@@ -177,6 +178,27 @@ var enum = func(column Column) string {
 	return withEnum(enum)(column)
 }
 
+var fileCache map[string][]string
```
nice! As the program works right now I think this is a cool approach.

It's interesting to notice that this is needed because the design of the program is still suboptimal. Right now, all the Constraints are computed for every single row which makes no sense from a perf perspective :D. I've been wanting to start working on it for a while (but only because it's fun!) but I had no real reason. Now that I see this I feel more compelled to start working on that too.

It was only a digression though (kind of a way for me to start thinking about this more concretely). I think what you did here is pretty good for the current design

jorinvo replied:
@lucapette sure, while adding this I was wondering about the same thing :D

jorinvo commented:
@lucapette I added it to the integration tests. 
Since the output is random, I introduced a new field in the table, `match`.
Please have a look if you would like to change something.

jorinvo commented on integration/cli_test.go:
```diff
@@ -109,6 +192,11 @@ func TestMain(m *testing.M) {
 		fmt.Printf("could not make binary for %s: %v", binaryName, err)
 		os.Exit(1)
 	}
+	binaryPath, err = filepath.Abs(binaryName)
```
Only need to get the path once.

jorinvo commented on integration/cli_test.go:
```diff
@@ -54,49 +56,130 @@ func TestCliArgs(t *testing.T) {
 	}
 }
 
+func TestCliErr(t *testing.T) {
```
These don't fit into the table above.

lucapette commented:
@jorinvo this landed on master with a different commit hash (didn't realise I needed to rebase before being able to push). I did some work on top of it as we discussed in private so that the tests are more stable and robust. I like this feature a lot. thank you very much for your help!
